src/Microsoft.AspNet.Http.Core/Extensions/MapWhenMiddleware.cs

```diff
@@ -20,27 +20,13 @@ public MapWhenMiddleware([NotNull] RequestDelegate next, [NotNull] MapWhenOption
 
         public async Task Invoke([NotNull] HttpContext context)
         {
-            if (_options.Predicate != null)
+            if (_options.Predicate(context))
```

Comment:
Hmm, true. Can we add a `[NotNull]` to the setter anyway? That way there would be no doubt.